src/Computers.java

```diff
@@ -0,0 +1,61 @@
+public class CalculatingDevice {
+    protected String name;
+
+    CalculatingDevice(String name) {
+        this.name = name;
+    }
+    
+    public String calculate(double x, double y) {
+        double sum = x + y;
+        return name + ": " + x + "+" + y + "=" + sum;
+    }
+
+    public static void printRes(CalculatingDevice[] a, double x, double y) {
+        for (int i = 0; i < a.length; i++) {
+            System.out.println(a[i].calculate(x, y));
+        }
+    }
+}
+
+public class Calculator extends CalculatingDevice {
+    public Calculator(String name) {
+        super(name);
+    }
+
+    @Override
+    public String calculate(double x, double y) {
+        String additionResult = super.calculate(x, y);
+        double sub = x - y;
+        return additionResult + "; " + x + "-" + y + "=" + sub;
```

Comment:
I think it will be more readable if you create a new function: `String subtraction(x, y)`, so that in this line you will have: `return addition + "; " + this.substraction(x, y);`.